RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -0,0 +1,102 @@
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "CondFormats/DataRecord/interface/BeamSpotObjectsRcd.h"
+//#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineLegacyObjectsRcd.h"
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineHLTObjectsRcd.h"
+#include "CondFormats/DataRecord/interface/BeamSpotTransientObjectsRcd.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "FWCore/Framework/interface/ESProductHost.h"
+#include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <memory>
+#include <iostream>
+#include <string>
+
+using namespace edm;
+
+class OnlineBeamSpotESProducer : public edm::ESProducer {
+public:
+  OnlineBeamSpotESProducer(const edm::ParameterSet& p);
+  ~OnlineBeamSpotESProducer() override;
+  std::shared_ptr<const BeamSpotObjects> produce(const BeamSpotTransientObjectsRcd&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& desc);
+
+private:
+  const BeamSpotOnlineObjects* compareBS(const BeamSpotOnlineObjects* bs1, const BeamSpotOnlineObjects* bs2);
+  BeamSpotObjects* fakeBS_;
+
+  edm::ESGetToken<BeamSpotObjects, BeamSpotTransientObjectsRcd> const bsToken_;
+  edm::ESGetToken<BeamSpotOnlineObjects, BeamSpotOnlineHLTObjectsRcd> bsHLTToken_;
+  edm::ESGetToken<BeamSpotOnlineObjects, BeamSpotOnlineLegacyObjectsRcd> bsLegacyToken_;
+};
+OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p) {
+  auto cc = setWhatProduced(this);
+
+  fakeBS_ = new BeamSpotOnlineObjects;
```

Comment:
per the smart pointer suggestion above: `std::move(std::make_unique< BeamSpotOnlineObjects >());`

Reply:
just for my education, why std::move() is needed here? Wouldn't make_unique be enough (just looking at other examples)?

Reply:
Correct, `std::move()` would not be needed here.

Reply:
Ah, thanks for correcting me, `std::move` is indeed redundant, this often trips me up. Reference: https://stackoverflow.com/questions/48322233/does-assigning-make-unique-require-stdmove-to-an-empty-unique-ptr